### PR TITLE
Rename auth failure listeners & add permission support

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
Renames auth failure listener classes & APIs to a more general name (rate limiters) to support API tokens & other use cases. Adds permission support to these APIs. Removes incorrect debug logging. 

* Category (Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation)
Enhancement
* Why these changes are required?
Fix: #4712
* What is the old behavior before changes and new behavior after changes?
New class names, API routes, permission support, before: none of the above. 

### Issues Resolved
Fix: #4712 

Is this a backport? If so, please add backport PR # and/or commits #, and remove `backport-failed` label from the original PR.
No

Do these changes introduce new permission(s) to be displayed in the static dropdown on the front-end? If so, please open a draft PR in the security dashboards plugin and link the draft PR here
No

### Testing
Existing tests pass
### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] New Roles/Permissions have a corresponding security dashboards plugin PR
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md)
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/security/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/security/dlic/rest/api/RestApiAdminPrivilegesEvaluator.java [hunk 1]
```diff
@@ -98,13 +99,6 @@ public boolean isCurrentUserAdminFor(final Endpoint endpoint, final String actio
             return false;
         }
         if (adminDNs.isAdmin(userAndRemoteAddress.getLeft())) {
-            if (logger.isDebugEnabled()) {
-                logger.debug(
-                    "Security admin permissions required for endpoint {} but {} is not an admin",
-                    endpoint,
-                    userAndRemoteAddress.getLeft().getName()
-                );
-            }
```
Comment (derek-ho) [inline src/main/java/org/opensearch/security/dlic/rest/api/RestApiAdminPrivilegesEvaluator.java:107]: I thought this was wrong so I removed it

File: src/main/java/org/opensearch/security/dlic/rest/api/RateLimitersApiAction.java [hunk 2]
```diff
@@ -75,18 +75,14 @@ public class AuthFailureListenersApiAction extends AbstractApiAction {
         )
     );
 
-    protected AuthFailureListenersApiAction(
```
Comment (derek-ho) [inline src/main/java/org/opensearch/security/dlic/rest/api/RateLimitersApiAction.java:78]: @cwperks @shikharj05 @DarshitChanpura what do you folks think about renaming this class to RateLimitersApiAction, but keeping the current routes as `/authfailurelisteners*`? This should give us the flexibility of later on adding other routes such as `/apitokenratelimiter*` to support other use cases?

File: src/main/java/org/opensearch/security/dlic/rest/api/RestApiAdminPrivilegesEvaluator.java [hunk 3]
```diff
@@ -66,6 +66,7 @@ default String build() {
         .put(Endpoint.CONFIG, action -> buildEndpointActionPermission(Endpoint.CONFIG, action))
         .put(Endpoint.INTERNALUSERS, action -> buildEndpointPermission(Endpoint.INTERNALUSERS))
         .put(Endpoint.NODESDN, action -> buildEndpointPermission(Endpoint.NODESDN))
+        .put(Endpoint.RATELIMITERS, action -> buildEndpointPermission(Endpoint.RATELIMITERS))
```
Comment (cwperks) [inline src/main/java/org/opensearch/security/dlic/rest/api/RestApiAdminPrivilegesEvaluator.java:69]: Will this also need updated in [AbstractApiValidationTest](https://github.com/opensearch-project/security/blob/main/src/test/java/org/opensearch/security/dlic/rest/api/AbstractApiActionValidationTest.java#L138) and [EndpointValidatorTest](https://github.com/opensearch-project/security/blob/main/src/test/java/org/opensearch/security/dlic/rest/validation/EndpointValidatorTest.java)?

File: src/main/java/org/opensearch/security/dlic/rest/api/RestApiAdminPrivilegesEvaluator.java (hunk 1 above)
Comment (willyborankin) [reply src/main/java/org/opensearch/security/dlic/rest/api/RestApiAdminPrivilegesEvaluator.java:107]: It was originally added for debugging purposes.

Comment (derek-ho) [reply src/main/java/org/opensearch/security/dlic/rest/api/RestApiAdminPrivilegesEvaluator.java:107]: Got it. What I meant is that the logic is inverted, the logging should state that the user is an admin.

**Review by willyborankin — APPROVED** [review]

**Review by cwperks — APPROVED** [review]

**Comment by willyborankin** [comment]
@derek-ho and @cwperks what about backport to 2.x?

**Pull request merged**